sim/midas/src/main/scala/midas/models/dram/Util.scala

```diff
@@ -265,6 +266,28 @@ trait HasAXI4Id extends HasNastiParameters { val id = UInt(nastiXIdBits.W) }
 trait HasAXI4IdAndLen extends HasAXI4Id { val len = UInt(nastiXLenBits.W) }
 trait HasReqMetaData extends HasAXI4IdAndLen { val addr = UInt(nastiXAddrBits.W) }
 
+class NastiAddressAttr (val addrBits: Int, val sizeBits: Int, val lenBits: Int)
+  (implicit val p: Parameters) extends Bundle { 
+  val addr = UInt(addrBits.W)
+  val size = UInt(sizeBits.W) 
+  val len = UInt(lenBits.W) 
+}
+
+class AddressLimits (val addrBits: Int) extends Bundle {
+  val addrmin = UInt(addrBits.W)
+  val addrmax = UInt(addrBits.W)
+}
+
+
+object AddressRangePair {
```

Comment:
Please add scala doc for this, and explain wether the arguments are going to be the AXI4 notions of `len` and `size`. It seems that way.

Reply:
Added comments.